Make install doesn't work due to rfc date standard issue in make file

**What happened**:
worked on : ubuntu and windows but not on mac and alpine based images
**make install** fails due to date issue on alpine based docker images and mac due to not supported rfc-3339 for date 
**What you expected to happen**:
It should install ocm on inside the image .
**How to reproduce it (as minimally and precisely as possible)**:
docker file where I have checked 
check in Makefile line no 17 
** NOW         := $(shell date --rfc-3339=seconds | sed 's/ /T/')**
**Anything else we need to know**:
logs :
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/cespare/xxhash/v2 v2.1.2
go: downloading github.com/prometheus/procfs v0.7.3
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.4
date: unrecognized option: rfc-3339=seconds
BusyBox v1.35.0 (2022-11-19 10:13:10 UTC) multi-call binary.

Usage: date [OPTIONS] [+FMT] [[-s] TIME]

Display time (using +FMT), or set time

        -u              Work in UTC (don't convert to local time)
        [-s] TIME       Set time to TIME
        -d TIME         Display TIME, not 'now'
        -D FMT          FMT (strptime format) for -s/-d TIME conversion
        -r FILE         Display last modification time of FILE
        -R              Output RFC-2822 date
        -I[SPEC]        Output ISO-8601 date
                        SPEC=date (default), hours, minutes, seconds or ns

Recognized TIME formats:
        @seconds_since_1970
        hh:mm[:ss]
        [YYYY.]MM.DD-hh:mm[:ss]
        YYYY-MM-DD hh:mm[:ss]
        [[[[[YY]YY]MM]DD]hh]mm[.ss]
        'date TIME' form accepts MMDDhhmm[[YY]YY][.ss] instead
date: unrecognized option: rfc-3339=seconds
BusyBox v1.35.0 (2022-11-19 10:13:10 UTC) multi-call binary.

Usage: date [OPTIONS] [+FMT] [[-s] TIME]

Display time (using +FMT), or set time

        -u              Work in UTC (don't convert to local time)
        [-s] TIME       Set time to TIME
        -d TIME         Display TIME, not 'now'
        -D FMT          FMT (strptime format) for -s/-d TIME conversion
        -r FILE         Display last modification time of FILE
        -R              Output RFC-2822 date
        -I[SPEC]        Output ISO-8601 date
                        SPEC=date (default), hours, minutes, seconds or ns

Recognized TIME formats:
        @seconds_since_1970
        hh:mm[:ss]
        [YYYY.]MM.DD-hh:mm[:ss]
        YYYY-MM-DD hh:mm[:ss]
        [[[[[YY]YY]MM]DD]hh]mm[.ss]
        'date TIME' form accepts MMDDhhmm[[YY]YY][.ss] instead
^Cmake: *** [Makefile:68: install] Interrupt

/opt/ocm # make install > logs.txt
date: unrecognized option: rfc-3339=seconds
BusyBox v1.35.0 (2022-11-19 10:13:10 UTC) multi-call binary.

Usage: date [OPTIONS] [+FMT] [[-s] TIME]

Display time (using +FMT), or set time

        -u              Work in UTC (don't convert to local time)      
        [-s] TIME       Set time to TIME
        -d TIME         Display TIME, not 'now'
        -D FMT          FMT (strptime format) for -s/-d TIME conversion
        -r FILE         Display last modification time of FILE
        -R              Output RFC-2822 date
        -I[SPEC]        Output ISO-8601 date
                        SPEC=date (default), hours, minutes, seconds or ns

Recognized TIME formats:
        @seconds_since_1970
        hh:mm[:ss]
        [YYYY.]MM.DD-hh:mm[:ss]
        YYYY-MM-DD hh:mm[:ss]
        [[[[[YY]YY]MM]DD]hh]mm[.ss]
        'date TIME' form accepts MMDDhhmm[[YY]YY][.ss] instead
date: unrecognized option: rfc-3339=seconds
**Environment**:

--- a/Makefile
+++ b/Makefile
@@ -14,7 +14,7 @@ COMPONENT_CLI_IMAGE_REPOSITORY                 := $(REGISTRY)/cli
 SOURCES := $(shell go list -f '{{$$I:=.Dir}}{{range .GoFiles }}{{$$I}}/{{.}} {{end}}' ./... )
 GOPATH                                         := $(shell go env GOPATH)
 
-NOW         := $(shell date -u +%FT%TZ)
+NOW         := $(shell date -u +%FT%T%:z)
 BUILD_FLAGS := "-s -w \
  -X github.com/open-component-model/ocm/pkg/version.gitVersion=$(EFFECTIVE_VERSION) \
  -X github.com/open-component-model/ocm/pkg/version.gitTreeState=$(GIT_TREE_STATE) \

--- a/Makefile
+++ b/Makefile
@@ -14,7 +14,7 @@ COMPONENT_CLI_IMAGE_REPOSITORY                 := $(REGISTRY)/cli
 SOURCES := $(shell go list -f '{{$$I:=.Dir}}{{range .GoFiles }}{{$$I}}/{{.}} {{end}}' ./... )
 GOPATH                                         := $(shell go env GOPATH)
 
-NOW         := $(shell date -u +%FT%T%:z)
+NOW         := $(shell date -u +%FT%T%z)
 BUILD_FLAGS := "-s -w \
  -X github.com/open-component-model/ocm/pkg/version.gitVersion=$(EFFECTIVE_VERSION) \
  -X github.com/open-component-model/ocm/pkg/version.gitTreeState=$(GIT_TREE_STATE) \

--- a/Makefile
+++ b/Makefile
@@ -14,7 +14,7 @@ COMPONENT_CLI_IMAGE_REPOSITORY                 := $(REGISTRY)/cli
 SOURCES := $(shell go list -f '{{$$I:=.Dir}}{{range .GoFiles }}{{$$I}}/{{.}} {{end}}' ./... )
 GOPATH                                         := $(shell go env GOPATH)
 
-NOW         := $(shell date --rfc-3339=seconds | sed 's/ /T/')
+NOW         := $(shell date -u +%FT%TZ)
 BUILD_FLAGS := "-s -w \
  -X github.com/open-component-model/ocm/pkg/version.gitVersion=$(EFFECTIVE_VERSION) \
  -X github.com/open-component-model/ocm/pkg/version.gitTreeState=$(GIT_TREE_STATE) \

--- a/components/demoplugin/Makefile
+++ b/components/demoplugin/Makefile
@@ -17,7 +17,7 @@ OCMSRCS=$(shell find $(REPO_ROOT)/pkg -type f) $(REPO_ROOT)/go.*
 GEN = $(REPO_ROOT)/gen/$(NAME)
 OCM = go run $(REPO_ROOT)/cmds/ocm
 
-NOW         := $(shell date -u +%FT%T%:z)
+NOW         := $(shell date -u +%FT%T%z)
 BUILD_FLAGS := "-s -w \
  -X github.com/open-component-model/ocm/pkg/version.gitVersion=$(EFFECTIVE_VERSION) \
  -X github.com/open-component-model/ocm/pkg/version.gitTreeState=$(GIT_TREE_STATE) \

--- a/components/demoplugin/Makefile
+++ b/components/demoplugin/Makefile
@@ -17,7 +17,7 @@ OCMSRCS=$(shell find $(REPO_ROOT)/pkg -type f) $(REPO_ROOT)/go.*
 GEN = $(REPO_ROOT)/gen/$(NAME)
 OCM = go run $(REPO_ROOT)/cmds/ocm
 
-NOW         := $(shell date --rfc-3339=seconds | sed 's/ /T/')
+NOW         := $(shell date -u +%FT%T%:z)
 BUILD_FLAGS := "-s -w \
  -X github.com/open-component-model/ocm/pkg/version.gitVersion=$(EFFECTIVE_VERSION) \
  -X github.com/open-component-model/ocm/pkg/version.gitTreeState=$(GIT_TREE_STATE) \

--- a/components/helminstaller/Dockerfile
+++ b/components/helminstaller/Dockerfile
@@ -1,6 +1,6 @@
 FROM --platform=$BUILDPLATFORM golang:1.19 AS builder
 ARG COMMIT EFFECTIVE_VERSION GIT_TREE_STATE
-ARG TARGETOS TARGETARCH 
+ARG TARGETOS TARGETARCH
 
 WORKDIR /go/src/github.com/open-component-model/ocm/
 COPY go.* ./
@@ -14,7 +14,7 @@ RUN --mount=type=cache,target=/root/.cache/go-build CGO_ENABLED=0 GOOS=$TARGETOS
 	-X github.com/open-component-model/ocm/pkg/version.gitVersion=$EFFECTIVE_VERSION \
 	-X github.com/open-component-model/ocm/pkg/version.gitTreeState=$GIT_TREE_STATE \
 	-X github.com/open-component-model/ocm/pkg/version.gitCommit=$COMMIT \
-	-X github.com/open-component-model/ocm/pkg/version.buildDate=$(date --rfc-3339=seconds | sed 's/ /T/')" \
+	-X github.com/open-component-model/ocm/pkg/version.buildDate=$(date -u +%FT%T%:z)" \
 	./cmds/helminstaller
 
 ###################################################################################

--- a/components/helminstaller/Dockerfile
+++ b/components/helminstaller/Dockerfile
@@ -14,7 +14,7 @@ RUN --mount=type=cache,target=/root/.cache/go-build CGO_ENABLED=0 GOOS=$TARGETOS
 	-X github.com/open-component-model/ocm/pkg/version.gitVersion=$EFFECTIVE_VERSION \
 	-X github.com/open-component-model/ocm/pkg/version.gitTreeState=$GIT_TREE_STATE \
 	-X github.com/open-component-model/ocm/pkg/version.gitCommit=$COMMIT \
-	-X github.com/open-component-model/ocm/pkg/version.buildDate=$(date -u +%FT%T%:z)" \
+	-X github.com/open-component-model/ocm/pkg/version.buildDate=$(date -u +%FT%T%z)" \
 	./cmds/helminstaller
 
 ###################################################################################

--- a/components/ocmcli/Makefile
+++ b/components/ocmcli/Makefile
@@ -18,7 +18,7 @@ OCMSRCS=$(shell find $(REPO_ROOT)/pkg -type f) $(REPO_ROOT)/go.*
 GEN = $(REPO_ROOT)/gen/$(shell basename $(realpath .))
 OCM = go run $(REPO_ROOT)/cmds/ocm
 
-NOW         := $(shell date -u +%FT%T%:z)
+NOW         := $(shell date -u +%FT%T%z)
 BUILD_FLAGS := "-s -w \
  -X github.com/open-component-model/ocm/pkg/version.gitVersion=$(EFFECTIVE_VERSION) \
  -X github.com/open-component-model/ocm/pkg/version.gitTreeState=$(GIT_TREE_STATE) \

--- a/components/ocmcli/Makefile
+++ b/components/ocmcli/Makefile
@@ -18,7 +18,7 @@ OCMSRCS=$(shell find $(REPO_ROOT)/pkg -type f) $(REPO_ROOT)/go.*
 GEN = $(REPO_ROOT)/gen/$(shell basename $(realpath .))
 OCM = go run $(REPO_ROOT)/cmds/ocm
 
-NOW         := $(shell date --rfc-3339=seconds | sed 's/ /T/')
+NOW         := $(shell date -u +%FT%T%:z)
 BUILD_FLAGS := "-s -w \
  -X github.com/open-component-model/ocm/pkg/version.gitVersion=$(EFFECTIVE_VERSION) \
  -X github.com/open-component-model/ocm/pkg/version.gitTreeState=$(GIT_TREE_STATE) \

--- a/hack/cross-build.sh
+++ b/hack/cross-build.sh
@@ -29,7 +29,7 @@ for i in "${build_matrix[@]}"; do
             -X github.com/open-component-model/ocm/pkg/version.gitVersion=$EFFECTIVE_VERSION \
             -X github.com/open-component-model/ocm/pkg/version.gitTreeState=$([ -z "$(git status --porcelain 2>/dev/null)" ] && echo clean || echo dirty) \
             -X github.com/open-component-model/ocm/pkg/version.gitCommit=$(git rev-parse --verify HEAD) \
-            -X github.com/open-component-model/ocm/pkg/version.buildDate=$(date --rfc-3339=seconds | sed 's/ /T/')" \
+            -X github.com/open-component-model/ocm/pkg/version.buildDate=$(date -u +%FT%T%:z)" \
   ${PROJECT_ROOT}/cmds/ocm
 
   # create zipped file

--- a/hack/cross-build.sh
+++ b/hack/cross-build.sh
@@ -29,7 +29,7 @@ for i in "${build_matrix[@]}"; do
             -X github.com/open-component-model/ocm/pkg/version.gitVersion=$EFFECTIVE_VERSION \
             -X github.com/open-component-model/ocm/pkg/version.gitTreeState=$([ -z "$(git status --porcelain 2>/dev/null)" ] && echo clean || echo dirty) \
             -X github.com/open-component-model/ocm/pkg/version.gitCommit=$(git rev-parse --verify HEAD) \
-            -X github.com/open-component-model/ocm/pkg/version.buildDate=$(date -u +%FT%T%:z)" \
+            -X github.com/open-component-model/ocm/pkg/version.buildDate=$(date -u +%FT%T%z)" \
   ${PROJECT_ROOT}/cmds/ocm
 
   # create zipped file

--- a/hack/install.sh
+++ b/hack/install.sh
@@ -19,5 +19,5 @@ CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) GO111MODULE=on \
             -X github.com/open-component-model/ocm/pkg/version.gitVersion=$EFFECTIVE_VERSION \
             -X github.com/open-component-model/ocm/pkg/version.gitTreeState=$([ -z git status --porcelain 2>/dev/null ] && echo clean || echo dirty) \
             -X github.com/open-component-model/ocm/pkg/version.gitCommit=$(git rev-parse --verify HEAD) \
-            -X github.com/open-component-model/ocm/pkg/version.buildDate=$(date -u +%FT%T%:z)" \
+            -X github.com/open-component-model/ocm/pkg/version.buildDate=$(date -u +%FT%T%z)" \
   ${PROJECT_ROOT}/cmds/ocm

--- a/hack/install.sh
+++ b/hack/install.sh
@@ -19,5 +19,5 @@ CGO_ENABLED=0 GOOS=$(go env GOOS) GOARCH=$(go env GOARCH) GO111MODULE=on \
             -X github.com/open-component-model/ocm/pkg/version.gitVersion=$EFFECTIVE_VERSION \
             -X github.com/open-component-model/ocm/pkg/version.gitTreeState=$([ -z git status --porcelain 2>/dev/null ] && echo clean || echo dirty) \
             -X github.com/open-component-model/ocm/pkg/version.gitCommit=$(git rev-parse --verify HEAD) \
-            -X github.com/open-component-model/ocm/pkg/version.buildDate=$(date --rfc-3339=seconds | sed 's/ /T/')" \
+            -X github.com/open-component-model/ocm/pkg/version.buildDate=$(date -u +%FT%T%:z)" \
   ${PROJECT_ROOT}/cmds/ocm

